documents/coredocument.go

```diff
@@ -287,6 +295,37 @@ func (cd *CoreDocument) getSigningRootProofHashes() (hashes [][]byte, err error)
 	return rootProof.SortedHashes, err
 }
 
+func (cd *CoreDocument) getSignatureDataSalts() ([]*coredocumentpb.DocumentSalt, error) {
```

Comment:
can we rename this as `setSignatureDataSalts` and not return the salts. Instead, we get the salts from the coredoc ? 
This will avoid any manipulation of salts once set to the document